src/fastq/call_consensus_reads.rs

```diff
@@ -258,47 +268,64 @@ pub fn call_consensus_reads_from_paths(
     umi_len: usize,
     seq_dist: usize,
     umi_dist: usize,
-) -> Result<(), Box<dyn Error>> {
+) -> Result<(), Context<FastqIOError>> {
     eprintln!("Reading input files:\n    {}\n    {}", fq1, fq2);
     eprintln!("Writing output to:\n    {}\n    {}", fq1_out, fq2_out);
 
     match (fq1.ends_with(".gz"), fq2.ends_with(".gz"), fq1_out.ends_with(".gz"), fq2_out.ends_with(".gz")) {
-        (false, false, false, false) => CallNonOverlappingConsensusRead::new(
-            &mut fastq::Reader::from_file(fq1).expect("Couldn't open first input file"),
-            &mut fastq::Reader::from_file(fq2).expect("Couldn't open second input file"),
-            &mut fastq::Writer::to_file(fq1_out).expect("Couldn't open first output file"),
-            &mut fastq::Writer::to_file(fq2_out).expect("Couldn't open second output file"),
+        (false, false, false, false) => match CallNonOverlappingConsensusRead::new(
+            &mut fastq::Reader::from_file(fq1).context(FastqIOError{ci_parameter: String::from("fq1"), path: String::from(fq1)})?,
+            &mut fastq::Reader::from_file(fq2).context(FastqIOError{ci_parameter: String::from("fq2"), path: String::from(fq2)})?,
+            &mut fastq::Writer::to_file(fq1_out).context(FastqIOError{ci_parameter: String::from("fq1_out"), path: String::from(fq1_out)})?,
+            &mut fastq::Writer::to_file(fq2_out).context(FastqIOError{ci_parameter: String::from("fq2_out"), path: String::from(fq1_out)})?,
             umi_len,
             seq_dist,
             umi_dist,
-        ).call_consensus_reads(),
-        (true, true, false, false) => CallNonOverlappingConsensusRead::new(
-            &mut fastq::Reader::new(fs::File::open(fq1).map(BufReader::new).map(GzDecoder::new).unwrap()),
-            &mut fastq::Reader::new(fs::File::open(fq2).map(BufReader::new).map(GzDecoder::new).unwrap()),
-            &mut fastq::Writer::to_file(fq1_out).expect("Couldn't open first output file"),
-            &mut fastq::Writer::to_file(fq2_out).expect("Couldn't open second output file"),
+        ).call_consensus_reads() {
+            Ok(()) => Ok(()),
+            // TODO the code below requires more elaborate handling
+            // The errors returned by rustbio do not implement Sync and Send
+            // If that would be the case, we could use
+            // call_consensus_reads().map_err(Error::from_boxed_compat)?
+            // TODO this is a Placeholder Error Type
```

Comment:
Ok, now that we already have this, we can also keep it for now and simplify later.